wait/http.go

```diff
@@ -173,38 +175,41 @@ func (ws *HTTPStrategy) WaitUntilReady(ctx context.Context, target StrategyTarge
 
 	var mappedPort nat.Port
 	if ws.Port == "" {
-		var err error
-		var ports nat.PortMap
 		// we wait one polling interval before we grab the ports otherwise they might not be bound yet on startup
-		for err != nil || ports == nil {
-			select {
-			case <-ctx.Done():
-				return fmt.Errorf("%w: %w", ctx.Err(), err)
-			case <-time.After(ws.PollInterval):
-				if err := checkTarget(ctx, target); err != nil {
-					return err
-				}
+		select {
+		case <-ctx.Done():
+			return fmt.Errorf("%w: %w", ctx.Err(), err)
+		case <-time.After(ws.PollInterval):
```

Comment:
So here we just wait and continue if context is not done, right? Can you help me understand the differences with checking the target inside the pollInterval case? 🙏

Reply:
Just reduced indentation, making the happy path the natural flow down the file, which typically reduces cognitive load, similar to how return errors early does.

The for loop was removed as the loop conditions were impossible, if an error was hit we return and ports is always set in the inspect output even if empty it seems.
```json
{
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "f196e8cebe297cab0a9a99130f218aec2d93bdc43d15cd9e3f53180f7c71feb9",
            "SandboxKey": "/var/run/docker/netns/f196e8cebe29",
            "Ports": {},
        }
}
```

Reply:
Added a comment for clarity.

Reply:
If the intent was to loop until ready, then loops needs to be restored and checked for `len(ports) == 0` and the comment corrected.

Reply:
Thanks for the explanations!!